pkg/minishift/hostfolder/hostfolder.go

```diff
@@ -0,0 +1,428 @@
+		return errors.New("No host folders defined.")
+	}
+
+	hostfolder := getHostfolderByName(name)
+	if hostfolder == nil {
+		return fmt.Errorf("No host folder defined as: %s", name)
+	} else {
+		ensureMountPointExists(driver, hostfolder)
+		mountHostfolder(driver, hostfolder)
+	}
+	return nil
+}
+
+// Performs mounting of host folders
+func MountHostfolders(driver drivers.Driver) error {
+	if !isHostRunning(driver) {
+		return errors.New("Host is in the wrong state.")
+	}
+
+	if !IsHostfoldersDefined(true) {
+		return errors.New("No host folders defined.")
+	}
+
+	println("-- Mounting hostfolders")
+
+	hostfolders := instanceState.Config.HostFolders
+	for i := range hostfolders {
+		// Ignore individual errors or aggregate
+		mountHostfolder(driver, &hostfolders[i])
+	}
+
+	return nil
+}
+
+func Umount(driver drivers.Driver, name string) error {
+	if !isHostRunning(driver) {
+		return errors.New("Host is in the wrong state.")
+	}
+
+	if !IsHostfoldersDefined(true) {
+		return errors.New("No host folders defined")
+	}
+
+	hostfolder := getHostfolderByName(name)
+	if hostfolder == nil {
+		return fmt.Errorf("No host folder defined as: %s", name)
+	} else {
+		umountHostfolder(driver, hostfolder)
+	}
+	return nil
+}
+
+func mountHostfolder(driver drivers.Driver, hostfolder *instanceState.HostFolder) error {
+	if hostfolder == nil {
+		return errors.New("Host folder not defined")
+	}
+
+	switch hostfolder.Type {
+	case "cifs":
+		if err := mountCifsHostfolder(driver, hostfolder); err != nil {
+			return err
+		}
+	default:
+		return errors.New("Unsupported host folder type")
+	}
+
+	return nil
+}
+
+func determineHostIp(driver drivers.Driver) (string, error) {
+	instanceip, err := driver.GetIP()
+	if err != nil {
+		return "", err
+	}
+
+	for _, hostaddr := range util.HostIPs() {
+
+		if util.NetworkContains(hostaddr, instanceip) {
+			hostip, _, _ := net.ParseCIDR(hostaddr)
+			if util.IsIPReachable(driver, hostip.String(), false) {
+				return hostip.String(), nil
+			} else {
+				return "", errors.New("Unreachable")
+			}
+		}
+	}
+
+	return "", errors.New("Unknown error occured")
+}
+
+func mountCifsHostfolder(driver drivers.Driver, hostfolder *instanceState.HostFolder) error {
+	// If "Users" is used as name, determine the IP of host for UNC path on startup
+	if hostfolder.Name == "Users" {
+		hostip, _ := determineHostIp(driver)
+		hostfolder.Options["uncpath"] = fmt.Sprintf("//%s/Users", hostip)
+	}
+
+	print(fmt.Sprintf("   Mounting '%s': '%s' as '%s' ... ",
+		hostfolder.Name,
+		hostfolder.Options["uncpath"],
+		hostfolder.Mountpoint()))
+
+	if isMounted, err := isHostfolderMounted(driver, hostfolder); isMounted {
+		println("Already mounted")
+		return fmt.Errorf("Host folder is already mounted. %s", err)
+	}
+
+	if !isCifsHostReachable(driver, hostfolder.Options["uncpath"]) {
+		print("Unreachable\n")
+		return errors.New("Host folder is unreachable")
+	}
+
+	cmd := fmt.Sprintf(
+		"sudo mount -t cifs %s %s -o username=%s,password=%s",
```

Comment:
or write in a credentials file and use `-o credentials=/tmp/hostfolder-[name].creds` instead.